CI OSX Conda build is flaky

The OSX Conda build sometimes fails. Here are some examples of recent failures:

https://github.com/gnuradio/gnuradio/actions/runs/2986963438/jobs/4812546038
https://github.com/gnuradio/gnuradio/actions/runs/2996076013/jobs/4806418288
https://github.com/gnuradio/gnuradio/actions/runs/2884399579/jobs/4583929663

Just before failing, the following is logged:

```
Partially parsed some '.tbd' files in sysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk, pretending .tbds are their install-names
Adding support to 'conda-build' for parsing these in 'liefldd.py' would be easy and useful:
['usr/lib/swift/libswiftSwiftOnoneSupport.tbd', 'System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd'] ...
WARNING:conda_build.post:Partially parsed some '.tbd' files in sysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk, pretending .tbds are their install-names
Adding support to 'conda-build' for parsing these in 'liefldd.py' would be easy and useful:
['usr/lib/swift/libswiftSwiftOnoneSupport.tbd', 'System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd'] ...
/Users/runner/miniforge3/etc/profile.d/conda.sh: line 9:  5271 Segmentation fault: 11  "$CONDA_EXE" $_CE_M $_CE_CONDA "$@"
/Users/runner/miniforge3/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
Error: Process completed with exit code 139.
```

The specific `.tbd` file being processed at the time differs, but the segmentation fault remains the same.

Unfortunately I don't know enough about Conda to diagnose this further.

--- a/.ci_support/linux_64_numpy1.21python3.10.____cpython.yaml
+++ b/.ci_support/linux_64_numpy1.21python3.10.____cpython.yaml
@@ -1,21 +1,23 @@
-MACOSX_DEPLOYMENT_TARGET:
-- '10.15'
-MACOSX_SDK_VERSION:
-- '10.15'
+alsa_lib:
+- 1.2.7
 boost_cpp:
-- 1.74.0
+- 1.78.0
 c_compiler:
-- clang
+- gcc
 c_compiler_version:
-- '13'
+- '10'
+cdt_name:
+- cos7
 channel_sources:
 - conda-forge
 channel_targets:
 - gnuradio main
 cxx_compiler:
-- clangxx
+- gxx
 cxx_compiler_version:
-- '13'
+- '10'
+docker_image:
+- quay.io/condaforge/linux-anvil-cos7-x86_64
 fftw:
 - '3'
 gmp:
@@ -24,28 +26,22 @@ gsl:
 - '2.7'
 libiio:
 - '0'
+libsndfile:
+- '1.1'
 libthrift:
 - 0.16.0
-macos_machine:
-- x86_64-apple-darwin13.4.0
 numpy:
-- '1.19'
+- '1.21'
 pin_run_as_build:
   boost-cpp:
     max_pin: x.x.x
-  fftw:
-    max_pin: x
-  gmp:
-    max_pin: x
   python:
     min_pin: x.x
-    max_pin: x.x
-  zeromq:
     max_pin: x.x
 pybind11_abi:
 - '4'
 python:
-- 3.9.* *_cpython
+- 3.10.* *_cpython
 qt_main:
 - '5.15'
 soapysdr:
@@ -53,9 +49,9 @@ soapysdr:
 spdlog:
 - '1.10'
 target_platform:
-- osx-64
+- linux-64
 uhd:
-- 4.2.0
+- 4.3.0
 volk:
 - '2.5'
 zeromq:

--- a/.ci_support/osx_64_numpy1.21python3.10.____cpython.yaml
+++ b/.ci_support/osx_64_numpy1.21python3.10.____cpython.yaml
@@ -1,35 +1,45 @@
+MACOSX_DEPLOYMENT_TARGET:
+- '10.9'
 boost_cpp:
-- 1.74.0
+- 1.78.0
 c_compiler:
-- vs2017
+- clang
+c_compiler_version:
+- '14'
 channel_sources:
 - conda-forge
 channel_targets:
 - gnuradio main
 cxx_compiler:
-- vs2017
+- clangxx
+cxx_compiler_version:
+- '14'
 fftw:
 - '3'
+gmp:
+- '6'
 gsl:
 - '2.7'
 libiio:
 - '0'
+libsndfile:
+- '1.1'
+libthrift:
+- 0.16.0
+macos_machine:
+- x86_64-apple-darwin13.4.0
 numpy:
-- '1.19'
+- '1.21'
 pin_run_as_build:
   boost-cpp:
     max_pin: x.x.x
-  fftw:
-    max_pin: x
   python:
     min_pin: x.x
     max_pin: x.x
-  zeromq:
-    max_pin: x.x.x
 pybind11_abi:
 - '4'
 python:
-- 3.9.* *_cpython
+- 3.10.* *_cpython
 qt_main:
 - '5.15'
 soapysdr:
@@ -37,13 +47,15 @@ soapysdr:
 spdlog:
 - '1.10'
 target_platform:
-- win-64
+- osx-64
 uhd:
-- 4.2.0
+- 4.3.0
 volk:
 - '2.5'
 zeromq:
 - 4.3.4
 zip_keys:
+- - c_compiler_version
+  - cxx_compiler_version
 - - python
   - numpy

--- a/.ci_support/win_64_numpy1.21python3.10.____cpython.yaml
+++ b/.ci_support/win_64_numpy1.21python3.10.____cpython.yaml
@@ -1,53 +1,33 @@
-alsa_lib:
-- 1.2.6
 boost_cpp:
-- 1.74.0
+- 1.78.0
 c_compiler:
-- gcc
-c_compiler_version:
-- '10'
-cdt_name:
-- cos7
+- vs2019
 channel_sources:
 - conda-forge
 channel_targets:
 - gnuradio main
 cxx_compiler:
-- gxx
-cxx_compiler_version:
-- '10'
-docker_image:
-- quay.io/condaforge/linux-anvil-cos7-x86_64
+- vs2019
 fftw:
 - '3'
-gmp:
-- '6'
 gsl:
 - '2.7'
-libcblas:
-- 3.9 *netlib
 libiio:
 - '0'
-libthrift:
-- 0.16.0
+libsndfile:
+- '1.1'
 numpy:
-- '1.19'
+- '1.21'
 pin_run_as_build:
   boost-cpp:
     max_pin: x.x.x
-  fftw:
-    max_pin: x
-  gmp:
-    max_pin: x
   python:
     min_pin: x.x
-    max_pin: x.x
-  zeromq:
     max_pin: x.x
 pybind11_abi:
 - '4'
 python:
-- 3.9.* *_cpython
+- 3.10.* *_cpython
 qt_main:
 - '5.15'
 soapysdr:
@@ -55,15 +35,13 @@ soapysdr:
 spdlog:
 - '1.10'
 target_platform:
-- linux-64
+- win-64
 uhd:
-- 4.2.0
+- 4.3.0
 volk:
 - '2.5'
 zeromq:
 - 4.3.4
 zip_keys:
-- - c_compiler_version
-  - cxx_compiler_version
 - - python
   - numpy

--- a/.github/workflows/conda-build.yml
+++ b/.github/workflows/conda-build.yml
@@ -13,22 +13,22 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - CONFIG: linux_64_python3.9.____cpython
-            SHORT_CONFIG: linux_64_python3.9.____cpython
+          - CONFIG: linux_64_numpy1.21python3.10.____cpython
+            SHORT_CONFIG: linux_64_numpy1.21python3.10.____cpython
             UPLOAD_PACKAGES: True
             DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
             os: ubuntu
-          - CONFIG: win_64_python3.9.____cpython
-            SHORT_CONFIG: win_64_python3.9.____cpython
+          - CONFIG: win_64_numpy1.21python3.10.____cpython
+            SHORT_CONFIG: win_64_numpy1.21python3.10.____cpython
             UPLOAD_PACKAGES: True
             os: windows
-          - CONFIG: osx_64_python3.9.____cpython
-            SHORT_CONFIG: osx_64_python3.9.____cpython
+          - CONFIG: osx_64_numpy1.21python3.10.____cpython
+            SHORT_CONFIG: osx_64_numpy1.21python3.10.____cpython
             UPLOAD_PACKAGES: True
             os: macos
     steps:
     - name: Checkout code
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
       with:
         fetch-depth: 0
 
@@ -84,11 +84,15 @@ jobs:
       shell: cmd
       run: |
         call activate base
-        mamba.exe install -c conda-forge 'python=3.9' conda-build conda pip boa conda-forge-ci-setup=3
+        mamba.exe install -c conda-forge 'python=3.9' conda-build conda pip boa conda-forge-ci-setup=3 "py-lief<0.12"
         if errorlevel 1 exit 1
         setup_conda_rc .\ ".\.packaging/conda_recipe" .\.ci_support\%CONFIG%.yaml
         if errorlevel 1 exit 1
         
+
+        if EXIST LICENSE.txt (
+          copy LICENSE.txt ".packaging/conda_recipe\\recipe-scripts-license.txt"
+        )
         conda.exe mambabuild ".packaging/conda_recipe" -m .ci_support\%CONFIG%.yaml
         if errorlevel 1 exit 1
         set "FEEDSTOCK_NAME=%GITHUB_REPOSITORY:*/=%"
@@ -146,7 +150,7 @@ jobs:
       continue-on-error: true
 
     - name: Store conda build artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v3
       if: ${{ always() && steps.prepare-artifacts.outcome == 'success' }}
       with:
         name: ${{ steps.prepare-artifacts.outputs.BLD_ARTIFACT_NAME }}
@@ -155,7 +159,7 @@ jobs:
       continue-on-error: true
 
     - name: Store conda build environment artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v3
       if: ${{ failure() && steps.prepare-artifacts.outcome == 'success' }}
       with:
         name: ${{ steps.prepare-artifacts.outputs.ENV_ARTIFACT_NAME }}

--- a/.packaging/conda_recipe/build.sh
+++ b/.packaging/conda_recipe/build.sh
@@ -1,5 +1,16 @@
 #!/usr/bin/env bash
 
+if [[ $target_platform == osx* ]] ; then
+    CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+fi
+
+# Workaround for no std::aligned_alloc with osx-64
+# https://github.com/chriskohlhoff/asio/issues/1090
+# Maybe remove when boost is updated to 1.80.0?
+if [[ "${target_platform}" == "osx-64" ]]; then
+  export CXXFLAGS="-DBOOST_ASIO_DISABLE_STD_ALIGNED_ALLOC ${CXXFLAGS}"
+fi
+
 cmake -E make_directory build
 cd build
 

--- a/.packaging/conda_recipe/conda_build_config.yaml
+++ b/.packaging/conda_recipe/conda_build_config.yaml
@@ -1,7 +1,2 @@
 channel_targets:
   - gnuradio main
-# need a minimum macOS that is higher than conda-forge default
-MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
-  - "10.15"                # [osx and x86_64]
-MACOSX_SDK_VERSION:        # [osx and x86_64]
-  - "10.15"                # [osx and x86_64]

--- a/.packaging/conda_recipe/meta.yaml
+++ b/.packaging/conda_recipe/meta.yaml
@@ -37,6 +37,7 @@ requirements:
     - ninja
     - pkg-config  # [not win]
     - sysroot_linux-64 2.17  # [linux64]
+    - thrift-compiler
     # cross-compilation requirements
     - python                              # [build_platform != target_platform]
     - cross-python_{{ target_platform }}  # [build_platform != target_platform]
@@ -65,8 +66,6 @@ requirements:
     - fftw
     - gmp  # [not win]
     - gsl
-    # blas needed to link with gsl
-    - libcblas  # [linux]
     - libsndfile
     - libthrift  # [not win]
     - mako
@@ -107,8 +106,10 @@ requirements:
     - cppzmq
     - zeromq
   # below needed only to run tests in host environment
+    - pytest
     - pyzmq
   run:
+    - alsa-plugins  # [linux]
     - boost-cpp
     - click
     - click-plugins

--- a/.packaging/conda_recipe/meta.yaml
+++ b/.packaging/conda_recipe/meta.yaml
@@ -18,7 +18,7 @@ source:
 
 build:
   number: 0
-  skip: true  # [py!=39]
+  skip: true  # [py!=310]
   entry_points:
     - gnuradio-companion = gnuradio.grc.main:main  # [win]
     - gr_filter_design = gnuradio.filter.filter_design:main  # [win]

--- a/.packaging/conda_recipe/meta.yaml
+++ b/.packaging/conda_recipe/meta.yaml
@@ -109,7 +109,6 @@ requirements:
   # below needed only to run tests in host environment
     - pyzmq
   run:
-    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
     - boost-cpp
     - click
     - click-plugins

--- a/.scripts/build_steps.sh
+++ b/.scripts/build_steps.sh
@@ -24,15 +24,18 @@ export CONFIG_FILE="${CI_SUPPORT}/${CONFIG}.yaml"
 cat >~/.condarc <<CONDARC
 
 conda-build:
- root-dir: ${FEEDSTOCK_ROOT}/build_artifacts
+  root-dir: ${FEEDSTOCK_ROOT}/build_artifacts
+pkgs_dirs:
+  - ${FEEDSTOCK_ROOT}/build_artifacts/pkg_cache
+  - /opt/conda/pkgs
 
 CONDARC
 
 
 mamba install --update-specs --yes --quiet --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3
+    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
 mamba update --update-specs --yes --quiet --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3
+    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
 
 # set up the condarc
 setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
@@ -53,6 +56,10 @@ make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
 
 ( endgroup "Configuring conda" ) 2> /dev/null
+
+if [[ -f "${FEEDSTOCK_ROOT}/LICENSE.txt" ]]; then
+  cp "${FEEDSTOCK_ROOT}/LICENSE.txt" "${RECIPE_ROOT}/recipe-scripts-license.txt"
+fi
 
 if [[ "${BUILD_WITH_CONDA_DEBUG:-0}" == 1 ]]; then
     if [[ "x${BUILD_OUTPUT_ID:-}" != "x" ]]; then

--- a/.scripts/run_osx_build.sh
+++ b/.scripts/run_osx_build.sh
@@ -23,11 +23,10 @@ bash $MINIFORGE_FILE -b -p ${MINIFORGE_HOME}
 source ${MINIFORGE_HOME}/etc/profile.d/conda.sh
 conda activate base
 
-echo -e "\n\nInstalling ['conda-forge-ci-setup=3'] and conda-build."
 mamba install --update-specs --quiet --yes --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3
+    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
 mamba update --update-specs --yes --quiet --channel conda-forge \
-    conda-build pip boa conda-forge-ci-setup=3
+    conda-build pip boa conda-forge-ci-setup=3 "py-lief<0.12"
 
 
 
@@ -56,6 +55,10 @@ source run_conda_forge_build_setup
 echo -e "\n\nMaking the build clobber file"
 make_build_number ./ ./.packaging/conda_recipe ./.ci_support/${CONFIG}.yaml
 
+
+if [[ -f LICENSE.txt ]]; then
+  cp LICENSE.txt ".packaging/conda_recipe/recipe-scripts-license.txt"
+fi
 
 if [[ "${BUILD_WITH_CONDA_DEBUG:-0}" == 1 ]]; then
     if [[ "x${BUILD_OUTPUT_ID:-}" != "x" ]]; then

